cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -137,6 +152,8 @@ void Run(const std::vector<string>& bag_filenames) {
     rosbag::View view(bag);
     const ::ros::Time begin_time = view.getBeginTime();
     const double duration_in_seconds = (view.getEndTime() - begin_time).toSec();
+    // The message processing loop publishes the clock.
```

Comment:
On the other hand, it might be easier if I leave it like it is now, so I don't have to worry about the code between timer.stop() and timer.start() such as serialization taking too long. Or can I put a timer.start right after the bag processing loop?